File: backend/app/schemas/article.py
from pydantic import BaseModel


class ArticleRead(BaseModel):
    id: int
    topic: str
    title: str
    source: str
    published_at: str
    summary: str
    url: str
    bookmarked: bool
    content: str = ""
    quality_score: int = -1
    quality_label: str = ""
    expires_at: str = ""

    model_config = {"from_attributes": True}


class ArticleBookmarkUpdate(BaseModel):
    bookmarked: bool

File: frontend/src/types/article.ts
// Article types
export interface Article {
  id: number
  topic: string
  title: string
  source: string
  published_at: string
  summary: string
  url: string
  bookmarked: boolean
  content: string
  quality_score: number
  quality_label: string
  expires_at: string
}

export interface IngestUrlResult {
  id: number | null
  title: string | null
  message: string
}

export interface CollectTopicResult {
  topic_name: string
  new_articles: number
  message: string
}
